src/ZLogger.Unity/Assets/Scripts/ZLogger/AsyncStreamLineMessageWriter.cs

```diff
@@ -190,7 +343,98 @@ await Task.Delay(sleepTime, cancellationTokenSource.Token)
+                    dropSummary[LogLevel.Error]       == 0 &&
+                    dropSummary[LogLevel.Critical]    == 0 &&
+                    dropSummary[LogLevel.None]        == 0)
+                {
+                    continue;
+                }
+
+                try
+                {
+                    CheckPostsTimesAndSetIsSpamming();
+                }
+                catch
+                {
+                    // ignored
+                }
+
+                if (isSpamming) continue;
+
+                // create summary log entry
+                try
+                {
+                    Exception? exception = null; // can add an exception new Exception("Log spamming detected")
+                    var logInfo = new LogInfo(
+                        0,
+                        "ZLogger",
+                        DateTimeOffset.Now,
+                        LogLevel.Critical,
+                        new EventId(0),
+                        exception
+                    );
+
+                    int droppedCount = dropSummary[LogLevel.Critical]
+                                     + dropSummary[LogLevel.Error]
+                                     + dropSummary[LogLevel.Warning]
+                                     + dropSummary[LogLevel.Information]
+                                     + dropSummary[LogLevel.Debug]
+                                     + dropSummary[LogLevel.Trace];
+
+                    IZLoggerEntry? entry =
+                        FormatLogState<object, int, int, int, int, int, int, int, uint, uint>.Factory(
+                            new
+                            (
+                                null,
+                                "Truncated {0} log messages (Critical: {1}, Error: {2}, Warning: {3}, Information: {4}, Debug: {5}, Trace: {6}) because had more than {7} logs in {8} seconds",
+                                droppedCount,
+                                dropSummary[LogLevel.Critical],
+                                dropSummary[LogLevel.Error],
+                                dropSummary[LogLevel.Warning],
+                                dropSummary[LogLevel.Information],
+                                dropSummary[LogLevel.Debug],
+                                dropSummary[LogLevel.Trace],
+                                LIMIT_IN,
+                                POSTS_SECONDS_WINDOW
+                            ),
+                            logInfo
+                        );
+
+                    channel.Writer.TryWrite(entry);
+
+                    // reset drop summary
+                    foreach (var key in dropSummary.Keys)
+                    {
+                        dropSummary[key] = 0;
+                    }
+                }
+                catch
+                {
+                    // ignored
+                }
+                finally
+                {
+                    didDrop = false;
```

Comment:
I don't really understand what `didDrop` represents (so maybe rename)

Reply:
It is for the use of `SummaryWriteLoop`
Also added lock for the read and write
960b2ef - lock didDrop bool